pkg/controller/notification/notification_controller.go

```diff
@@ -166,12 +186,46 @@ func (r *ReconcileNotification) setStatusNotificationDeletionFailed(notification
 		})
 }
 
-func (r *ReconcileNotification) setStatusNotificationSent(notification *toolchainv1alpha1.Notification) error {
+func (r *ReconcileNotification) setStatusNotificationContextError(notification *toolchainv1alpha1.Notification, msg string) error {
 	return r.updateStatusConditions(
 		notification,
 		toolchainv1alpha1.Condition{
-			Type:   toolchainv1alpha1.NotificationSent,
-			Status: corev1.ConditionTrue,
-			Reason: toolchainv1alpha1.NotificationSentReason,
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationContextErrorReason,
+			Message: msg,
 		})
 }
+
+func (r *ReconcileNotification) setStatusNotificationDeliveryError(notification *toolchainv1alpha1.Notification, msg string) error {
+	return r.updateStatusConditions(
+		notification,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationDeliveryErrorReason,
```

Comment:
i guess we will need to raise a PR with these reasons added? looks like there is an old PR that can be re-purposed? https://github.com/codeready-toolchain/api/pull/129/files

Reply:
The API module already contains the new reasons as far as I'm aware.. i'm not sure what that old draft PR is, perhaps I should close it.

Reply:
ohhhh okay i see it :+1: